packages/frontend/@n8n/chat/src/components/MessageActions.vue

```diff
@@ -0,0 +1,67 @@
+<script setup lang="ts">
+import { N8nTooltip, N8nIcon } from '@n8n/design-system';
+
+import { useChat, useOptions } from '@n8n/chat/composables';
+import { chatEventBus } from '@n8n/chat/event-buses';
+import type { ChatMessage } from '@n8n/chat/types';
+
+const props = defineProps<{
+	message: ChatMessage;
+}>();
+
+const { options } = useOptions();
+const chatStore = useChat();
+
+async function repostMessage() {
+	if (props.message.sender === 'user') {
+		// Repost user message by sending it again
+		const messageText =
+			'text' in props.message && typeof props.message.text === 'string' ? props.message.text : '';
+		if (messageText.trim()) {
+			await chatStore.sendMessage(
+				messageText,
+				props.message.files ? Array.from(props.message.files) : [],
+			);
+		}
+	}
+}
+
+function copyToInput() {
+	const messageText =
+		'text' in props.message && typeof props.message?.text === 'string' ? props.message?.text : '';
+	if (messageText.trim()) {
+		chatEventBus.emit('setInputValue', messageText);
+	}
+}
+</script>
+
+<template>
+	<div v-if="options.enableMessageActions" class="message-actions">
+		<N8nTooltip v-if="message.sender === 'user'">
+			<N8nIcon icon="redo-2" size="medium" class="icon" @click="repostMessage" />
+			<template #content> Resend message </template>
+		</N8nTooltip>
+		<N8nTooltip v-if="message.sender === 'user'">
+			<N8nIcon icon="files" size="medium" class="icon" @click="copyToInput" />
+			<template #content> Reuse Message </template>
```

Comment:
we have i18n values configured in packages/frontend/@n8n/chat/src/constants/defaults.ts